src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/Directory/GetFiles.cs

```diff
@@ -192,5 +192,40 @@ public void DirectoryWithTrailingSeparators(string trailing)
             string[] files = GetEntries(root + (IsDirectoryInfo ? trailing : ""), "*", SearchOption.AllDirectories);
             FSAssert.EqualWhenOrdered(new string[] { rootFile, nestedFile }, files);
         }
+
+        [Theory, MemberData(nameof(TestData.ValidFileNames), MemberType = typeof(TestData))]
+        public void EnumerateFilesWithProblematicNames(string fileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            File.Create(Path.Combine(testDir.FullName, fileName)).Dispose();
+
+            string[] files = GetEntries(testDir.FullName);
+            Assert.Single(files);
+            Assert.Contains(files, f => Path.GetFileName(f) == fileName);
+        }
+
+        [ConditionalTheory(nameof(UsingNewNormalization))]
+        [MemberData(nameof(TestData.WindowsTrailingProblematicFileNames), MemberType = typeof(TestData))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void WindowsEnumerateFilesWithTrailingSpacePeriod(string fileName)
+        {
+            // Files with trailing spaces/periods must be created with \\?\ on Windows
+            // but enumeration can find them. This tests the scenario from issue #113120.
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string filePath = Path.Combine(testDir.FullName, fileName);
+            File.Create(@"\\?\" + filePath).Dispose();
+
+            string[] files = GetEntries(testDir.FullName);
+            Assert.Single(files);
+            Assert.Contains(files, f => Path.GetFileName(f) == fileName);
+            
+            // The problematic scenario: enumeration returns the filename with trailing space/period,
+            // but File.Exists and File.OpenRead will fail on the returned path because
+            // normalization strips the trailing characters.
+            // This is tracked in issue #113120.
```

Comment:
This does not match the scenario from #113120. In order to reproduce the scenario from #113120, you have to:
1. Create a directory with a name from WindowsTrailingProblematicFileNames set
2. Create a file with non-problematic name in that directory
3. Enumerate the directory

Expected behavior: directory enumerator can return valid full path to the file with non-problematic name   
Actual behavior: directory enumerator does not return valid full path

You may want to create a new test function for this